src/Compilers/Core/Portable/Syntax/GreenNode.NodeEnumerable.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis;
+
+internal abstract partial class GreenNode
+{
+    public ref struct NodeEnumerable(GreenNode node)
```

Comment:
What makes these need to be ref structs?

Reply:
It's less a need and more a preference.  Helps avoid thigns like accidentally boxing, using linq, etc.  basically it's a mechanism to say: this really should exist only on the stack (and basically, only as a foreach) :)  

I can take taht out.  But i've been a big fan of having objects effectively being explicit that they're intended to be short-lived, stack-only values :)

Reply:
I was more assuming you'd have marked the structs with `NonCopyable`. I don't hate the idea of keeping them ref structs until we find a use case though.

Reply:
i can do that too :)

Reply:
> Helps avoid thigns like accidentally boxing, using linq, etc

This is completely unnecessary with `[NonCopyable]`, and artificially prevents the type from working in async code. `ref struct` should not be used as a replacement for `NonCopyable`.